### PR TITLE
Add the token text to the fly_success page.

### LINKED ISSUE #4914: Allow copying auth token even when browser copy/paste is disabled
## What challenge are you facing?

I'm supporting an operations team using concourse. The organization has the policy to disable automated copy/paste in the browser. Copying using the 'manual' (ctrl+c / right-click + copy) methods still work as expected. This means that when logging in to fly on a remote machine, two things happen:

1. The browser is unable to send the token to fly automatically
2. The user is unable to click the copy token to clipboard button

Our current workaround is to right-click+inspect the 'copy token to clipboard' button and copy the token out of the `data-clipboard-text` attribute.

Screenshot from 2019-12-17 13-51-06

Also of note, we checked the value of `ClipboardJS.isSupported()` and it returns `true`, so it doesn't seem easy to automatically determine if the feature is blocked. We have validated this behavior on Chrome and Firefox.

## What would make this better?

We think it would be valuable to somehow display the token in a box so it can be highlighted and copied manually by the user. This would mirror the approach taken for example on GitHub with repository remote urls:

Screenshot from 2019-12-17 13-52-42

## Are you interested in implementing this yourself?

We're open to submitting a PR if there is a path to getting it accepted.

### PR DESCRIPTION
**Pull request by jvshahid**
Some customers have security policies to disable automated copy/paste in the browser.  Copying using the 'manual' (ctrl+c / right-click + copy) methods still work as expected.  This change allows the user to manually copy the token.

# Existing Issue

Fixes #4914.

# Changes proposed in this pull request

* This change allows the user to manually copy the token, by adding the token text to the `fly_success` page

# Reviewer Checklist
- [x] Code reviewed
- [x] Tests reviewed
- [ ] Documentation reviewed
- [ ] Release notes reviewed
- [x] PR acceptance performed
- [ ] New config flags added? Ensure that they are added to the [BOSH](https://github.com/concourse/concourse-bosh-release) 
      and [Helm](https://github.com/concourse/helm) packaging; otherwise, ignored for the [integration tests](https://github.com/concourse/ci/tree/master/tasks/scripts/check-distribution-env) (for example, if they are Garden configs that are not displayed in the `--help` text).

### CODE REVIEW

**Comment by jvshahid** [comment]
cc @Logiraptor

**Comment by jvshahid** [comment]
Attached is a screenshot of the `fly_success` page after the change we made.  We intentionally did not put effort into the design and will wait for guidance.


Screenshot from 2019-12-27 14-41-21

**Comment by xtremerui** [comment]
@Lindsayauchin @pivotal-jamie-klassen any suggestion of presenting the token in the login successful page? Or we could just go with the Github style mentioned in the original issue? Thx!

**Comment by matthewpereira** [comment]
@Lindsayauchin maybe we can pair on the UI for this? I have a [draft idea](https://www.figma.com/proto/fuoFSVcTGptiS21P2POA2XbN/Concourse-UI-kit-and-Experiments?node-id=466%3A4&scaling=min-zoom) to start if you want to take a look.

**Comment by Lindsayauchin** [comment]
How about this?

Screen Shot 2020-02-18 at 2 27 02 PM

**Comment by Logiraptor** [comment]
@Lindsayauchin That looks perfect for us.

**Comment by zoetian** [comment]
@Logiraptor Hey Patrick, are you still interested in implementing this feature?

**Comment by zoetian** [comment]
> @jvshahid How about this?
> 
> Screen Shot 2020-02-18 at 2 27 02 PM

Hey @Lindsayauchin , just a small clarification:

Will the first token always be truncated (maybe first 16 characters with dots); the only way to get full text is to click on the copied text?

Thanks!

**Comment by zoetian** [comment]
Seems like what we want here is `text-flow: ellipsis;` 
Also, cavet: https://stackoverflow.com/a/17783233/3693070

**Comment by jvshahid** [comment]
> @Logiraptor Hey Patrick, are you still interested in implementing this feature?

Hey Zoe, yes we are still interested in this feature.

**Comment by Logiraptor** [comment]
Hi @zoetian, I just finally got a chance to push the style changes requests. I've attached a before and after.

# Before

Screen Shot 2020-07-16 at 4 13 10 PM

# After

Screen Shot 2020-07-16 at 4 12 41 PM

Can you let me know if this looks good to merge at this point? Thanks!

**Comment by zoetian** [comment]
@Logiraptor Heyyyy, this looks amazing! Can you please create a PR on the concourse repo for it? Thanks!

**Comment by Logiraptor** [comment]
@zoetian Awesome! Sorry, should I create a new PR besides this one?

**Comment by zoetian** [comment]
@Logiraptor Sorry, my mistake... I was viewing the issue page. You don't need to create a new PR. My bad, my bad.

File: web/elm/src/FlySuccess/FlySuccess.elm [hunk 1]
```diff
@@ -14,7 +14,7 @@ import FlySuccess.Models as Models exposing (ButtonState(..), Model, hover)
 import FlySuccess.Styles as Styles
 import FlySuccess.Text as Text
 import Html exposing (Html)
-import Html.Attributes exposing (attribute, href, id, style)
+import Html.Attributes exposing (attribute, class, href, id, style, value)
```
Comment (zoetian) [inline web/elm/src/FlySuccess/FlySuccess.elm:17]: Heyyy, if you run the `yarn analyse -s` and go to `localhost:3000`, you will see there are two error messages; here we need to delete the unused `class`.

File: web/elm/src/FlySuccess/FlySuccess.elm [hunk 2]
```diff
@@ -184,13 +184,26 @@ body model =
...
+
+
+tokenTextBox : String -> Html Message
+tokenTextBox token =
+    Html.label []
+        [ Html.text Text.copyTokenInput
+        , Html.input
+            ([ value token
+             ]
+                ++ Styles.input
```
Comment (zoetian) [inline web/elm/src/FlySuccess/FlySuccess.elm:203]:
```suggestion
            (value token
                :: Styles.input
```

**Comment by Logiraptor** [comment]
@zoetian Thanks for the review! I fixed the issues reported by `yarn analyze`, added the hover behavior you mentioned, and added tests to cover it all. Let me know if anything else is needed to get this merged. 😄

**Comment by zoetian** [comment]
Hey @Logiraptor, somehow the DCO failed here :( Do you mind fixing this one? Thank you! =)

**Comment by jomsie** [comment]
@zoetian I just re-kicked off testflight containerd since I saw it red!

**Pull request merged**